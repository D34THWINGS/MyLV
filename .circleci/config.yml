build_common: &build_common
  executor:
    name: node/node
    tag: '10'
  working_directory: ~/mylv
  environment:
    NODE_ENV: production
  steps:
  - attach_workspace:
      at: ~/
  - run: yarn build
  - store_artifacts:
      path: ~/mylv/dist
  - store_artifacts:
      path: ~/mylv/packages/emails/dist
  - store_artifacts:
      path: ~/mylv/packages/documents/dist
  - persist_to_workspace:
      root: ~/
      paths:
      - mylv

deploy_common: &deploy_common
  docker:
  - image: buildpack-deps:trusty
  working_directory: ~/mylv
  steps:
  - attach_workspace:
      at: ~/
  - run: git checkout -b release
  - run: git add -f dist/ packages/emails/dist/ packages/documents/dist/
  - run: git config user.email "deploy@circleci.com"
  - run: git config user.name "CircleCI Deployer"
  - run: git commit -m "CircleCI deploy"
  - deploy:
      command: git push -f https://heroku:$HEROKU_API_KEY@git.heroku.com/$HEROKU_APP.git release:master

tag_filters: &tag_filters
  tags:
    only: /^v\d+\.\d+\.\d+/

version: 2.1

orbs:
  node: circleci/node@0.0.2

jobs:
  install:
    executor:
      name: node/node
      tag: '10'
    working_directory: ~/mylv
    steps:
    - checkout
    - node/with-cache:
        steps:
          - run: yarn install --frozen-lockfile
        cache-key: yarn.lock
    - persist_to_workspace:
        root: ~/
        paths:
          - mylv
          - .cache/Cypress

  build_dev:
    <<: *build_common
    environment:
      BABEL_ENV: dev

  build_staging:
    <<: *build_common
    environment:
      BABEL_ENV: staging

  build_production:
    <<: *build_common
    environment:
      BABEL_ENV: production

  test:
    executor:
      name: node/node
      tag: '10'
    working_directory: ~/mylv
    steps:
    - attach_workspace:
        at: ~/
    - run: yarn lint
    - run: yarn test -w 8

#  test_e2e:
#    docker:
#    - image: cypress/base:10
#    - image: mongo:4.0
#    working_directory: ~/mylv
#    environment:
#      APP_ENV: test
#    steps:
#    - attach_workspace:
#        at: ~/
#    - run:
#        command: yarn start
#        background: true
#    - run:
#        command: yarn e2e:record
#        no_output_timeout: 1m

  hold_deployment:
    docker:
    - image: buildpack-deps:trusty
    steps:
    - run: echo "Waiting for approval before deployement"

  deploy_staging:
    <<: *deploy_common
    environment:
      HEROKU_APP: mylv-staging

  deploy_production:
    <<: *deploy_common
    environment:
      HEROKU_APP: mylv

workflows:
  version: 2
  general_workflow:
    jobs:
    - install:
        filters:
          branches:
            ignore: master
          tags:
            ignore: /.*/
    - build_dev:
        requires:
          - install
    - test:
        requires:
        - build_dev
#    - test_e2e:
#        requires:
#        - build_dev

  master_workflow:
    jobs:
    - install:
        filters:
          branches:
            only: master
          tags:
            ignore: /.*/
    - build_staging:
        requires:
          - install
    - test:
        requires:
        - build_staging
#    - test_e2e:
#        requires:
#        - build_staging
    - deploy_staging:
        requires:
        - test
#        - test_e2e

  release_workflow:
    jobs:
    - install:
        filters:
          branches:
            ignore: /.*/
          tags:
            only: /^v\d+\.\d+\.\d+/
    - build_production:
        requires:
          - install
        filters: *tag_filters
    - hold_deployment:
        type: approval
        requires:
        - build_production
        filters: *tag_filters
    - deploy_production:
        requires:
        - hold_deployment
        filters: *tag_filters
