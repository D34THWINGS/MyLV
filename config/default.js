module.exports = {
  server: {},
  host: {
    hostname: null,
    port: 8001,
  },
  logs: {
    reporters: {
      consoleReporter: [{
        module: 'good-console',
        args: [{ log: '*', response: '*', worker: '*' }],
      }, 'stdout'],
    },
  },
  mongodb: {
    host: 'localhost',
    port: 27017,
    database: 'cracra',
    config: {
      useMongoClient: true,
    },
  },
  lvconnect: {
    appId: null,
    appSecret: null,
    endpoint: 'https://lvconnect.link-value.fr',
  },
  cracra: {
    lunchesRoles: ['rh', 'board', 'business'],
    partnersRoles: ['rh', 'board'],
  },
}
