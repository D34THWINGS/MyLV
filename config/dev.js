module.exports = {
  host: {
    hostname: 'localhost',
  },
  front: {
    url: 'http://localhost:3000',
    featureFlipping: {
      holidays: true,
      transport: true,
      offlineMode: true,
      pushNotifications: true,
    },
  },
}
