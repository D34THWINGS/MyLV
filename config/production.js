function reportingFactory(path, options) {
  return [{
    module: 'good-squeeze',
    name: 'Squeeze',
    args: [options],
  }, {
    module: 'good-squeeze',
    name: 'SafeJson',
  }, {
    module: 'good-file',
    args: [path],
  }]
}

module.exports = {
  host: {
    port: process.env.PORT || 8001,
  },
  logs: {
    reporters: {
      accessReporting: reportingFactory('./logs/hapi/hapi-out.log', { log: '*', response: '*' }),
      errorReporting: reportingFactory('./logs/hapi/hapi-error.log', { error: '*' }),
      workerReporting: reportingFactory('./logs/hapi/hapi-worker.log', { worker: '*' }),
    },
  },
  mongodb: {
    uri: process.env.MONGODB_URI,
  },
  lvconnect: {
    appId: process.env.LVCONNECT_APP_ID,
    appSecret: process.env.LVCONNECT_APP_SECRET,
  },
  front: {
    featureFlipping: {
      offlineMode: true,
      pushNotifications: true,
    },
    sentry: {
      dsn: '12183e96fee94f34bf55a85d0fe1c802',
      projectId: '418084',
    },
  },
}
