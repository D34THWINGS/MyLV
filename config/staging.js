function reportingFactory(path, options) {
  return [{
    module: 'good-squeeze',
    name: 'Squeeze',
    args: [options],
  }, {
    module: 'good-squeeze',
    name: 'SafeJson',
  }, {
    module: 'good-file',
    args: [path],
  }]
}

module.exports = {
  host: {
    port: process.env.PORT || 8001,
  },
  logs: {
    reporters: {
      accessReporting: reportingFactory('./logs/hapi/hapi-out.log', { log: '*', response: '*' }),
      errorReporting: reportingFactory('./logs/hapi/hapi-error.log', { error: '*' }),
      workerReporting: reportingFactory('./logs/hapi/hapi-worker.log', { worker: '*' }),
    },
  },
  mongodb: {
    uri: process.env.MONGODB_URI,
  },
  lvconnect: {
    appId: process.env.LVCONNECT_STAGING_APP_ID,
    appSecret: process.env.LVCONNECT_STAGING_APP_SECRET,
  },
}
