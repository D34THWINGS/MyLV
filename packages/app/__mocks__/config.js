module.exports = {
  appName: 'FooApp',
  lvconnect: {},
  cracra: {},
}
