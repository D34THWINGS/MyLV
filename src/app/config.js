module.exports = process.env.CONFIG
