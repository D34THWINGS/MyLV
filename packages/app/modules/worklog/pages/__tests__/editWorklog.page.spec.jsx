import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import config from '@cracra/config/app'

import { EditWorklogPage } from '../editWorklog.page'

jest.mock('@cracra/config/app', () => ({ lvconnect: {} }))
jest.unmock('../editWorklog.page')

describe('WorklogPage', () => {
  let props

  const getWrapper = () => shallow(<EditWorklogPage {...props} />)

  beforeEach(() => {
    props = {
      classes: {},
      shouldRemindProcess: false,
      canPrint: true,
      isTabletOrMobile: false,
      isOffline: false,
    }

    config.featureFlipping = {
      transport: false,
    }
  })

  it('should render properly', () => {
    // When
    const wrapper = getWrapper()

    // Then
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('should redirect if missing client info', () => {
    // Given
    props.canPrint = false

    // When
    const wrapper = getWrapper()

    // Then
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('should display process reminder', () => {
    // Given
    props.shouldRemindProcess = true

    // When
    const wrapper = getWrapper()

    // Then
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
