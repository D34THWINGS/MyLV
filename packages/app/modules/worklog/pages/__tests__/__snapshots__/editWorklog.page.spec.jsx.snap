// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorklogPage should display process reminder 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Mon CRA | 
    </title>
  </HelmetWrapper>
  <WithStyles(Grid)
    container={true}
    spacing={16}
  >
    <Connect(Process) />
    <WithStyles(Grid)
      item={true}
      md={4}
      xs={12}
    >
      <Connect(WithStyles(ReduxForm)) />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      md={8}
      xs={12}
    >
      <Connect(WithStyles(Calendar)) />
    </WithStyles(Grid)>
    <WithStyles(Fab)
      className=""
      color="primary"
      onClick={[Function]}
    >
      <pure(PrintIcon) />
    </WithStyles(Fab)>
  </WithStyles(Grid)>
  <Connect(WithStyles(Printer)) />
</Fragment>
`;

exports[`WorklogPage should redirect if missing client info 1`] = `
<Redirect
  push={false}
  to="/client"
/>
`;

exports[`WorklogPage should render properly 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Mon CRA | 
    </title>
  </HelmetWrapper>
  <WithStyles(Grid)
    container={true}
    spacing={16}
  >
    <WithStyles(Grid)
      item={true}
      md={4}
      xs={12}
    >
      <Connect(WithStyles(ReduxForm)) />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      md={8}
      xs={12}
    >
      <Connect(WithStyles(Calendar)) />
    </WithStyles(Grid)>
    <WithStyles(Fab)
      className=""
      color="primary"
      onClick={[Function]}
    >
      <pure(PrintIcon) />
    </WithStyles(Fab)>
  </WithStyles(Grid)>
  <Connect(WithStyles(Printer)) />
</Fragment>
`;
